cmd/gateway-gcs.go

```diff
@@ -1036,9 +1046,26 @@ func (l *gcsGateway) CompleteMultipartUpload(bucket string, key string, uploadID
 	}
 
 	var parts []*storage.ObjectHandle
-	for _, uploadedPart := range uploadedParts {
+	partSizes := make([]int64, len(uploadedParts))
+	for i, uploadedPart := range uploadedParts {
 		parts = append(parts, l.client.Bucket(bucket).Object(gcsMultipartDataName(uploadID,
 			uploadedPart.PartNumber, uploadedPart.ETag)))
+		partAttr, pErr := l.client.Bucket(bucket).Object(gcsMultipartDataName(uploadID, uploadedPart.PartNumber, uploadedPart.ETag)).Attrs(l.ctx)
```

Comment:
> Object returns an ObjectHandle, which provides operations on the named object. This call does not perform any network operations.

A network call to fetch object attributes is made when `ObjectHandle.Attrs` method is called. So, there is only one call to the backend here.

Reply:
Looks like there is an extra backend call here - the `i`-th part is being fetched twice (before the call to `.Attrs()`). The result from the first call should be reused.